subprojects/platform-jvm/src/main/java/org/gradle/api/java/archives/internal/DefaultAttributes.java

```diff
@@ -48,35 +80,40 @@ public boolean containsValue(Object value) {
 
     @Override
     public Object get(Object key) {
+        if (key instanceof String) {
+            return attributes.get(keyFor((String) key));
+        }
         return attributes.get(key);
     }
 
     @Override
     public Object put(String key, Object value) {
         if (key == null) {
-            throw new ManifestException("The key of a manifest attribute must not be null.");
+            // NPE is the specified exception type (see Map#put javadoc)
+            throw new NullPointerException("The key of a manifest attribute must not be null.");
         }
         if (value == null) {
-            throw new ManifestException(String.format("The value of a manifest attribute must not be null (Key=%s).", key));
+            // NPE is the specified exception type (see Map#put javadoc)
```

Comment:
❌ Let's remove the comment here, as noted, we should comply with the `Map` contract.